rocAL/source/readers/image/file_source_reader.cpp

```diff
@@ -52,6 +52,7 @@ Reader::Status FileSourceReader::initialize(ReaderConfig desc) {
     auto ret = Reader::Status::OK;
     _file_id = 0;
     _folder_path = desc.path();
+    _file_list_path = desc.json_path();
```

Comment:
You should not get this from Json path, add another variable and API in reader config class to get the file list path